addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/WebRequester.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hpprinter.internal.handler;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.NoRouteToHostException;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.hpprinter.internal.config.HPPrinterConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+/**
+ * Performs web requests to EWS.
+ *
+ * @author Stewart Cossey - Initial contribution
+ */
+@NonNullByDefault
+public class WebRequester {
+    public static final String REQ_STATUS_OK = "ok";
+    public static final String REQ_STATUS_ENDPOINTFAIL = "fail";
+    public static final String REQ_STATUS_DOCEXCEPTION = "exception";
+
+    private final Logger logger = LoggerFactory.getLogger(HPPrinterHandler.class);
+
+    HPPrinterConfiguration config;
+    String endpoint;
+
+    private final int connectTimeout = 6000;
+    private final int readTimeout = 10000;
+
+    public WebRequester(String endpoint, HPPrinterConfiguration config) {
+        this.endpoint = endpoint;
+        this.config = config;
+    }
+
+    public boolean check() {
+        HttpsURLConnection checkSecureConnection = null;
```

Comment:
`HttpURLConnection` has `HttpURLConnection` parent class knowing this you can remove a lot of duplication, please take a general look into that.

Eg for this method:

```java
HttpURLConnection checkConnection = null;
try {
    URL url = new URL(config.useSSL ? "https" : "http", config.ipAddress, endpoint);
    checkConnection = (HttpURLConnection) url.openConnection();
    checkConnection.setConnectTimeout(connectTimeout);
    checkConnection.setReadTimeout(readTimeout);
    checkConnection.connect();
    return true;
} catch (SocketTimeoutException | NoRouteToHostException | IOException e) {
    logger.debug("An error occurred while connecting", e);
    return false;
} finally {
    if (checkConnection != null) {
        checkConnection.disconnect();
    }
}